bpf/types.h

```diff
@@ -124,14 +121,17 @@ typedef struct flow_id_t {
 // Force emitting struct flow_id into the ELF.
 const struct flow_id_t *unused2 __attribute__((unused));
 
-// Standard 4 tuple and a sequence identifier.
+// Standard 4 tuple, transport protocol and a sequence identifier.
 // No need to emit this struct. It's used only in kernel space
 typedef struct flow_seq_id_t {
     u16 src_port;
     u16 dst_port;
     u8 src_ip[IP_MAX_LEN];
     u8 dst_ip[IP_MAX_LEN];
     u32 seq_id;
+    u8 transport_protocol;
+    u32 if_index; // OS interface index
+    u8 __padding;
```

Comment:
no need to add manual padding attribute will takecare of it, also seq_id is becoming more like `flow_id` I feel we can use the global hash map to calculate RTT directly none tcp fields like icmp ones can be set to 0

Reply:
I added `transport_protocol` to make it future proof, to address your comment of seq id collision. and to make it similar to the `dns seq_id` to combine them in future if needed. 
Padding is something I can remove. 
if index is needed because flow agent connects to many interfaces in the same path which can affect time stamp when SYN and ACK were recorded to a. large extent if the same flow passes through eth0 -> bridge -> container veth -> container namespace ethernet etc

Reply:
yeah its just more and more becoming similar to flow_id

Reply:
There won't be more additions to it in my opinion. This map also stores seq_id which is the primary identifier here which does not have to be part of flow_id.